### PR TITLE
3.x firmware compatible, speedx, set_analog_out and new_csys_from_xpy

### PR DESCRIPTION
**Pull request by alvcap**
- ursecmon.py: Parses packages from 3.x firmware. Still could be improved for making urx able to handle all firmwares.
- speedx in robot.py: avoid orienting speed when only joint movement is desired (speedj)
- set_analog_out: send correct URScript program
- new_csys_from_xpy: fine tuned

### CODE REVIEW

File: urx/robot.py [hunk 1]
```diff
@@ -149,9 +149,13 @@ def speedx(self, command, velocities, acc, min_time):
...
-        v = self.csys.orient * m3d.Vector(velocities[:3])
-        w = self.csys.orient * m3d.Vector(velocities[3:])
-        URRobot.speedx(self, command, np.concatenate((v.array, w.array)), acc, min_time)
+        if command != "speedj":
+            v = self.csys.orient * m3d.Vector(velocities[:3])
+            w = self.csys.orient * m3d.Vector(velocities[3:])
+            vels = np.concatenate((v.array, w.array))
+        else:
+            vels = velocities
+        URRobot.speedx(self, command, vels, acc, min_time)
```
Comment (oroulet) [inline urx/robot.py]: It is probably cleaner to not call speedx from speed instead of using if for 90% of method

File: urx/robot.py [hunk 2]
```diff
@@ -214,20 +218,26 @@ def new_csys_from_xpy(self):
         Return new coordinate system from three points: X, Origin, Y
         based on math3d: Transform.new_from_xyp
         """
+        # Set coord. sys. to 0
+        self.csys = m3d.Transform()
```
Comment (oroulet) [inline urx/robot.py]: If you reset csys you need to restore it after!

Comment (alvcap) [inline urx/robot.py]: I agree, how should we do it?

a) Restore csys inside the method, and don't return the result
b) Restore csys inside the method, and return the result
c) Return result as now, and specify in docstring that user must restore csys afterwards

File: urx/ursecmon.py [hunk 3]
```diff
@@ -88,7 +88,9 @@ def parse(self, data):
...
                 allData["LaserPointer(OBSOLETE)"] = self._get_data(pdata, "iBddd", ("size", "type"))
             elif ptype == 3:
-                allData["MasterBoardData"] = self._get_data(pdata, "iBhhbbddbbddffffBBb", ("size", "type", "digitalInputBits", "digitalOutputBits", "analogInputRange0", "analogInputRange1", "analogInput0", "analogInput1", "analogInputDomain0", "analogInputDomain1", "analogOutput0", "analogOutput1", "masterBoardTemperature", "robotVoltage48V", "robotCurrent", "masterIOCurrent"))  # , "masterSafetyState" ,"masterOnOffState", "euromap67InterfaceInstalled"   ))
+                #fmt = "iBhhbbddbbddffffBBb"    # firmware 1.6, 1.7, 1.8
+                fmt = "iBiibbddbbddffffBBb"     # firmware 3.0, 3.1, 3.2
+                allData["MasterBoardData"] = self._get_data(pdata, fmt, ("size", "type", "digitalInputBits", "digitalOutputBits", "analogInputRange0", "analogInputRange1", "analogInput0", "analogInput1", "analogInputDomain0", "analogInputDomain1", "analogOutput0", "analogOutput1", "masterBoardTemperature", "robotVoltage48V", "robotCurrent", "masterIOCurrent"))  # , "masterSafetyState" ,"masterOnOffState", "euromap67InterfaceInstalled"   ))
```
Comment (oroulet) [inline urx/ursecmon.py]: we need to add an "if" there. version is known

**Pull request merged**